src/amo/components/Link.js

```diff
@@ -9,37 +9,75 @@ import { Link } from 'react-router';
 
 export class LinkBase extends React.Component {
   static propTypes = {
-    base: PropTypes.string,
+    clientApp: PropTypes.string.isRequired,
     children: PropTypes.node,
     href: PropTypes.string,
+    lang: PropTypes.string.isRequired,
+    prependClientApp: PropTypes.bool,
+    prependLang: PropTypes.bool,
     to: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
   }
 
   static defaultProps = {
-    base: '',
+    prependClientApp: true,
+    prependLang: true,
+  }
+
+  urlPrefix({ clientApp, lang, prependClientApp, prependLang } = {}) {
+    if (prependClientApp && prependLang) {
+      return `/${lang}/${clientApp}/`;
+    } else if (prependClientApp) {
+      return `/${clientApp}/`;
+    } else if (prependLang) {
+      return `/${lang}/`;
+    }
```

Comment:
This could be DRY'd up to:
```js
const prefix = [];
if (prependLang) {
  prefix.push(lang);
}
if (prependClientApp) {
  prefix.push(clientApp);
}
if (prefix.length) {
  return `/${prefix.join('/')}/`;
}
return null;
```